### PR TITLE
chore: initial e2e tests

### PR DESCRIPTION
**Pull request by tracy-french**
**What this PR does / why we need it**: This change includes the initial e2e test setup and a confirmation test for the "Get property value" query (see https://github.com/grafana/iot-sitewise-datasource/issues/212). The initial implementation does not rely on a provisioned data source and instead mocks network requests. 

The change also includes a GitHub action for running the tests on PR push (following guidance int Grafana testing guide).

Follow-up PRs will include:
* Tests for all query types (including non-happy path tests)
* Tests for the config editor
* Tests for the asset browser
* Increased test writing scalability (e.g., mocking requests, test fixtures, utilities)
* True e2e tests using real SiteWise resources and services

The test added assumes the fix in https://github.com/grafana/iot-sitewise-datasource/pull/286 is merged to test asset property selection.

### CODE REVIEW

**Comment by tracy-french** [comment]
Updated PR with GH action.

**Comment by tracy-french** [comment]
@kevinwcyu Thank you for your time and feedback! I will work on addressing all of your comments shortly!

Exciting to be one of the first using `@grafana/plugin-e2e`. I'll write up a few of my thoughts about the package and drop them here for you.

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    outputs:
+      matrix: ${{ steps.resolve-versions.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Resolve Grafana E2E versions
+        id: resolve-versions
+        uses: grafana/plugin-actions/e2e-version@main
+        with:
+          skip-grafana-dev-image: true
```
[the remark this answers is not in the document]
Comment (tracy-french) [reply .github/workflows/e2e.yml]: I get a build failure on the dev image: https://github.com/grafana/iot-sitewise-datasource/actions/runs/8144706762/job/22259389974?pr=285#step:10:36

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        id: run-tests
+        run: yarn playwright test
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v4
+        if: ${{ (always() && steps.run-tests.outcome == 'success') || (failure() && steps.run-tests.outcome == 'failure') && github.event.organization.login != 'grafana' }}
+        with:
+          name: playwright-report-${{ matrix.GRAFANA_IMAGE.NAME }}-v${{ matrix.GRAFANA_IMAGE.VERSION }}-${{github.run_id}}
+          path: playwright-report/
+          retention-days: 30
```
[the remark this answers is not in the document]
Comment (tracy-french) [reply .github/workflows/e2e.yml]: This job appears to require additional configuration on the repo? https://github.com/grafana/iot-sitewise-datasource/actions/runs/8145135150/job/22260712602?pr=285#step:13:28.

**Comment by tracy-french** [comment]
@kevinwcyu All of your feedback has been addressed. Just running into the issues with the dev image and GCS.

File: .github/workflows/e2e.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevinwcyu) [reply .github/workflows/e2e.yml]: I believe the [`docker-compose.yaml`](https://github.com/grafana/iot-sitewise-datasource/blob/main/docker-compose.yaml) needs to be updated to include the `grafana_image` as an argument.

It should look like the following snippet.

```yaml
version: '3.0'

services:
  grafana:
    container_name: 'grafana-iot-sitewise-datasource'
    platform: 'linux/amd64'
    build:
      context: ./.config
      args:
        grafana_image: ${GRAFANA_IMAGE:-grafana-enterprise}
        grafana_version: ${GRAFANA_VERSION:-9.2.5}
    ports:
      - 3000:3000/tcp
    volumes:
      - ./dist:/var/lib/grafana/plugins/grafana-iot-sitewise-datasource
      - ./provisioning:/etc/grafana/provisioning
```

File: .github/workflows/e2e.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevinwcyu) [reply .github/workflows/e2e.yml]: Looks like an issue with the GH action being run from a fork. I've asked for some clarification on this from the publish report action author. I'll let you know what the suggested action when they get back to me.

**Review by kevinwcyu — APPROVED** [review]
LGTM. Tentatively approving as it should be ready once the `docker-compose.yaml` file is updated and we figure out what to do with the publish report step (this will be on our end to figure out).

Nice catch on the `Format` property having an incorrect `id`.

File: .github/workflows/e2e.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tracy-french) [reply .github/workflows/e2e.yml]: This did the trick on the dev image! Thanks a ton, @kevinwcyu!

**Comment by tracy-french** [comment]
@kevinwcyu As the tests are now running on every version and we just have the GCS action left to handle, I'm going to merge this. Maybe we'll see the action succeed 🤞.

**Pull request merged**